### ISSUE TITLE
Standalone CSS is Broken

### ISSUE THREAD

**Issue by Recon419A** [naturalcrit/homebrewery#847]
The standalone CSS file provided in the project root, and the one created by running `npm run phb`, break expected syntax and do not work across a variety of tools, including Atom. Testing with multiple web formatting tools and several methods in Atom, nothing appears able to parse the provided format; other CSS files from other sources are confirmed to work using the same methods.

**Comment by ericscheid** [comment]
This one? https://raw.githubusercontent.com/naturalcrit/homebrewery/master/phb.standalone.css

“It's an older code sir, but it checks out.”

Translation: the CSS appears to valid (per a couple of validators), but it is also out of date (i.e. changes to live css haven't been copied to here too).

**Comment by calculuschild** [comment]
I don't notice any problems with the one in the project root. Can you specify the errors you seem to be getting?

**Comment by ericscheid** [comment]
The standalone.css has various instances of `column-break-inside:avoid;` which have been replaced with `page-break-inside:avoid; break-inside:avoid;`, insertions of `z-index: 150;` (or 100 etc), the .classTable has changed from `border-image-repeat:round;` to `border-image-repeat:stretch;` etc etc. standalone.css is out of date (by about 2 years?)

Hmm ... standalone.css also includes @font-face rules where the src property is defined as

    src:url(data:font/opentype;base64,T1RUTwAN [...] JoADA==) format('opentype');

Maybe the lack of quotes around that URL is confusing things? The W3C spec example doesn't have the src url quoted, but then that example also doesn't have a semicolon in it. I could see some parsers choking on that.

**Comment by Recon419A** [comment]
It's refusing to highlight as valid CSS in my editor (Atom) and doesn't work when trying to use it with, for example, the `markdown-themable-pdf` package to create PDFs from markdown in Atom, or with Atom's built-in CSS style sheet for the PDF previews.

**Comment by Recon419A** [comment]
It does seem to be the URLs that are breaking things

**Comment by Recon419A** [comment]
Specifically, it keeps telling me `expected ')', got ''` on all the lines with base64 URLs

**Comment by ericscheid** [comment]
Two things need doing:

1. bring the standalone css up to date
2. quotify the urls in the code

**Comment by G-Ambatte** [comment]
Looks like the standalone CSS file hasn't changed in over 5 years at this point, and the PHB generation script doesn't work either.

My assumption is that the Legacy theme is the appropriate CSS; updating line 4 of the `phb.js` script to `less.render(fs.readFileSync('./themes/5ePhbLegacy.style.less', 'utf8'), { compress: true })` causes the script to work again; the output file appears to contain all of the `.phb` declarations and all base64 URIs appear to be quotified, so I believe it is a 1:1 replacement for the original `phb.standalone.css`.

We could also link the `phb.js` script to the `buildall` script to ensure that the `phb.standalone.css` file is always kept up to date, in the event that there are future changes to the Legacy styles.

**Comment by calculuschild** [comment]
I would probably say the V3 theme should be the "standard" CSS file since it's the one receiving updates and is more "accurate" to the actual book. They will also need fonts, etc. (although images are embedded into the CSS) to make this stuff work, since we have explicitly removed them from the CSS.

But it does bring up the question about what people are using this for. Would users want *all* the themes available as we add them?

**Comment by G-Ambatte** [comment]
As I understand it, anyone who is or has been using `phb.standalone.css` will have their documents set up to use `.phb` divs; changing to the v3 theme will require them to change to `.page` divs, which is why I think we should stick to Legacy CSS.

It does raise the question of "Is this a feature that people are even using?" This particular issue has been open for almost three years now, and as @ericscheid pointed out at the very start, the standalone CSS hadn't been updated in over two years at that time.

My thinking is that we make `./phb.standalone.css` the CSS file of the compiled Legacy Less, for anyone out there who has been using it... We then mark it as a Legacy feature and provide minimal support for anyone continuing to use it, it would only be updated if/when the Legacy theme is changed - most likely, it never gets touched again.

---

I just ran a quick test, importing the Legacy CSS into Atom, as that seems to be the use case of the original reporter (@Recon419A)  and the compressed CSS still appears to have issues. I re-ran the output with `compress: false` and that produced a CSS file with less issues, however base64 URLs still appears to have issues.

![image](https://user-images.githubusercontent.com/19826659/148215075-4c77b2e3-6dbc-4815-ba3d-d17fd71795fe.png)

That said, my exposure to Atom has consisted of an hour or two this evening, so it's also entirely possible that I have made a mistake here somehow.
